crates/ruff_linter/src/rules/ruff/rules/unnecessary_regular_expression.rs

```diff
@@ -0,0 +1,243 @@
+use ruff_diagnostics::{AlwaysFixableViolation, Diagnostic, Edit, Fix};
+use ruff_macros::{derive_message_formats, ViolationMetadata};
+use ruff_python_ast::{
+    Arguments, CmpOp, Expr, ExprAttribute, ExprCall, ExprCompare, ExprContext, Identifier,
+};
+use ruff_python_semantic::Modules;
+use ruff_text_size::TextRange;
+
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+///
+/// Reports the following `re` calls when their first arguments are plain string
+/// literals, and no additional flags are passed:
+///
+/// - `sub`
+/// - `match`
+/// - `search`
+/// - `fullmatch`
+/// - `split`
+///
+/// For `sub`, the `repl` (replacement) argument must also be a string literal,
+/// not a function. For `match`, `search`, and `fullmatch`, the return value
+/// must also be used only for its truth value.
+///
+/// ## Why is this bad?
+///
+/// Performing checks on strings directly can make the code simpler, may require
+/// less escaping, and will often be faster.
+///
+/// ## Example
+///
+/// ```python
+/// re.sub("abc", "", s)
+/// ```
+///
+/// Use instead:
+///
+/// ```python
+/// s.replace("abc", "")
+/// ```
+#[derive(ViolationMetadata)]
+pub(crate) struct UnnecessaryRegularExpression {
+    replacement: String,
+}
+
+impl AlwaysFixableViolation for UnnecessaryRegularExpression {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        "Plain string pattern passed to `re` function".to_string()
+    }
+
+    fn fix_title(&self) -> String {
+        format!("Replace with `{}`", self.replacement)
+    }
+}
+
+/// The `re` functions supported by this rule.
+#[derive(Debug)]
+enum ReFuncKind<'a> {
+    Sub { repl: &'a Expr },
+    Match,
+    Search,
+    Fullmatch,
+    Split,
+}
+
+#[derive(Debug)]
+struct ReFunc<'a> {
+    kind: ReFuncKind<'a>,
+    pattern: &'a Expr,
+    string: &'a Expr,
+}
+
+impl<'a> ReFunc<'a> {
+    fn from_call_expr(
+        checker: &'a mut Checker,
+        call: &'a ExprCall,
+        func_name: &str,
+    ) -> Option<Self> {
+        let nargs = call.arguments.len();
+        let locate_arg = |name, position| call.arguments.find_argument(name, position);
```

Comment:
Nit: I personally found the alias more confusing than helpful. Calling `call.arguments.find_argument` isn't much longer. You could consider storing `arguments` in a variable, if the length annoys you

```
let arguments = &call.arguments;
arguments.find_argument("pattern", 0)?
```